lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -501,105 +565,60 @@ class StableHLOToTTIRReduceWindowOpConversionPattern
-              .getShape();
-
-      // Dead ttir.constant sticks around and fails verification. Removing it
-      // like so since its behind another op
-      recursiveErase(rewriter, adaptor.getInitValues()[0].getDefiningOp());
-      rewriter.replaceOpWithNewOp<mlir::tt::ttir::MaxPool2dOp>(
-          srcOp, outputType, srcOp.getInputs()[0], outputTensor,
-          kernel_height_attr, kernel_width_attr, stride_height_attr,
-          stride_width_attr, dilation_height_attr, dilation_width_attr,
-          rewriter.getBoolAttr(false), rewriter.getSI32IntegerAttr(padding[0]),
-          rewriter.getSI32IntegerAttr(padding[1]),
-          rewriter.getSI32IntegerAttr(padding[2]),
-          rewriter.getSI32IntegerAttr(padding[3]),
-          rewriter.getArrayAttr(
-              SmallVector<Attribute>(adaptor.getOperands().size() + 1,
-                                     rewriter.getAttr<OperandConstraintAttr>(
-                                         OperandConstraint::AnyDeviceTile))),
-          rewriter.getSI32IntegerAttr(input_shape[1]),
-          rewriter.getSI32IntegerAttr(input_shape[2]));
-
-      return success();
+      outputs_vec.push_back(outputTensor);
     }
-    return failure();
+    ValueRange outputs = outputs_vec;
+
+    auto window_dimensions = adaptor.getWindowDimensionsAttr();
+    auto window_strides = adaptor.getWindowStridesAttr();
+    auto base_dilations = adaptor.getBaseDilationsAttr();
+    auto window_dilations = adaptor.getWindowDilationsAttr();
+    auto padding_ = adaptor.getPaddingAttr();
+
+    // Generate defaults if they dont exist
+    window_strides =
+        window_strides ? window_strides
+                       : rewriter.getDenseI64ArrayAttr(
+                             SmallVector<int64_t>(window_dimensions.size(), 1));
+    base_dilations =
+        base_dilations ? base_dilations
+                       : rewriter.getDenseI64ArrayAttr(
+                             SmallVector<int64_t>(window_dimensions.size(), 1));
+    window_dilations = window_dilations
+                           ? window_dilations
+                           : rewriter.getDenseI64ArrayAttr(SmallVector<int64_t>(
+                                 window_dimensions.size(), 1));
+    auto padding =
+        padding_ ? rewriter.getDenseI64ArrayAttr(
+                       SmallVector<int64_t>(padding_.getValues<int64_t>()))
+                 : rewriter.getDenseI64ArrayAttr(
+                       SmallVector<int64_t>(window_dimensions.size() * 2, 1));
+
+    auto operand_constraints = rewriter.getArrayAttr(SmallVector<Attribute>(
+        adaptor.getOperands().size(), rewriter.getAttr<OperandConstraintAttr>(
+                                          OperandConstraint::AnyDeviceTile)));
+
+    mlir::tt::ttir::PoolingMethod pooling_method;
+    if (isMaxPool(srcOp)) {
+      pooling_method = mlir::tt::ttir::PoolingMethod::Max;
+    } else {
+      return failure();
```

Comment:
If we make it this far down in the rewriter, maybe we should fail with an error message: `rewriter.notifyMatchFailure(op, "Unsupported pooling method")`.